consensus/src/chained_bft/network.rs

```diff
@@ -34,43 +33,11 @@ use std::{
     time::{Duration, Instant},
 };
 
-/// The response sent back from EventProcessor for the BlockRetrievalRequest.
+/// The block retrieval request is used internally for implementing RPC: the callback is executed
+/// for carrying the response
 #[derive(Debug)]
-pub struct BlockRetrievalResponse<T> {
-    pub status: BlockRetrievalStatus,
-    pub blocks: Vec<Block<T>>,
-}
-
-impl<T: Payload> BlockRetrievalResponse<T> {
-    pub fn verify(&self, block_id: HashValue, num_blocks: u64) -> failure::Result<()> {
-        ensure!(
-            self.status != BlockRetrievalStatus::Succeeded
-                || self.blocks.len() as u64 == num_blocks,
-            "not enough blocks returned, expect {}, get {}",
-            num_blocks,
-            self.blocks.len(),
-        );
-        self.blocks
-            .iter()
-            .try_fold(block_id, |expected_id, block| {
-                ensure!(
-                    block.id() == expected_id,
-                    "blocks doesn't form a chain: expect {}, get {}",
-                    expected_id,
-                    block.id()
-                );
-                Ok(block.parent_id())
-            })
-            .map(|_| ())
-    }
-}
-
-/// BlockRetrievalRequest carries a block id for the requested block as well as the
-/// oneshot sender to deliver the response.
-#[derive(Debug)]
-pub struct BlockRetrievalRequest<T> {
-    pub block_id: HashValue,
-    pub num_blocks: u64,
+pub struct BlockRetrievalRequestWithCallback<T> {
```

Comment:
I know the name literally describes the fields, but not describe the purpose imho, sth like `IncomingBlockRetrievalRequest`, but up to you.